pkg/build/golang/utils.go

```diff
@@ -0,0 +1,45 @@
+package golang
+
+import (
+	"strings"
+)
+
+func parseDependencies(raw string) map[string]string {
+	rawModules := strings.Split(raw, "\n")
+	modules := map[string]string{}
+
+	for _, module := range rawModules {
+		module = strings.TrimSpace(module)
+		if module == "" {
+			continue
+		}
+
+		parts := strings.Split(module, " ")
+		path := strings.TrimSpace(parts[0])
+		version := ""
+
+		if len(parts) == 2 {
+			version = strings.TrimSpace(parts[1])
+		}
+
+		modules[path] = version
+	}
+
+	return modules
+}
+
+func parseDependenciesFromDocker(raw string) map[string]string {
```

Comment:
With piping the dependency graph into a file inside the image, all of this disappears. Instead I'd create a container (without starting it), and I'd use `docker cp` (API equivalent) to copy the dependency file out. Then delete the container.

Reply:
Or just run a container with `--rm` and override the entrypoint to `cat`... this way no need to `docker cp`, and it'd just be one command. Just an idea.

Reply:
I basically went with @nonsense's idea.

Reply:
As you prefer. The reason I suggested a `docker cp` is because the Docker API doesn't have a single command for `docker run`. You actually have to call create, **then** start, and actually process the output. It's clunky. And starting a container implies a higher latency.

Since you _anyway_ have to create the container, you can just extract the file without starting it and doing all the above dance.

Reply:
@raulk missed this comment! Yes, I will try to do that to see if it's faster (it must be) :)